x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -452,8 +452,24 @@ static Settings additionalSettings(final Settings settings, final boolean enable
                 new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(ReadOnlyAction.NAME), ReadOnlyAction::parse),
                 new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(RolloverAction.NAME), RolloverAction::parse),
                 new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(ShrinkAction.NAME), ShrinkAction::parse),
-                new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(DeleteAction.NAME), DeleteAction::parse)
-        );
+                new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(DeleteAction.NAME), DeleteAction::parse),
+                // ILM - Client - Lifecycle Actions
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.DeleteAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.DeleteAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.DeleteAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.ForceMergeAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.ForceMergeAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.ForceMergeAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.ReadOnlyAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.ReadOnlyAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.ReadOnlyAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.RolloverAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.RolloverAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.RolloverAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.ShrinkAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.ShrinkAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.ShrinkAction::parse)
```

Comment:
Same comment here as in the allocate action PR about needing a common `categoryClass` for all the actions so we can deserialise them without knowing the exact action type and also same comment about moving the  registration of the server side objects away from the client.